src/languages/json-source-code.js

```diff
@@ -107,9 +138,27 @@ export class JSONSourceCode extends TextSourceCodeBase {
 	constructor({ text, ast }) {
 		super({ text, ast });
 		this.ast = ast;
-		this.comments = ast.tokens
-			? ast.tokens.filter(token => token.type.endsWith("Comment"))
-			: [];
+	}
+
+	/**
+	 * Ensures that the tokens and comments are processed and cached.
+	 * @returns {void}
+	 */
+	#ensureTokens() {
+		if (!this.#comments) {
+			const { comments, starts } = processTokens(this.ast.tokens ?? []);
+			this.#comments = comments;
+			this.#tokenStarts = starts;
+		}
+	}
```

Comment:
ESLint will always call some of the methods that use comments, so I think this isn't necessary and that we can move this to the constructor to avoid calling `this.#ensureTokens()` in every method that needs tokens or comments.